Automattic/jetpack
# Protect: Threat history improvements

Fixes #
- Improved History display error state structure/styling
- Add Scanner/History access points to error/scanning states
- Add filter access to History display when filter has no results
- Updates the text for the empty list to specify the filter applied

### Other information:

## Jetpack product discussion



## Does this pull request change what data or activity we track or use?


- No

## Testing instructions:






* Checkout this branch and run it on JT
* Manually set the `error` property to `true` in the `useProtectData` hook and view both the Scanner and History displays, verify that the error state is reasonable
* Ensure that you can access History from the Scanner in error and scanning state and the Scanner from the History in the error state
* View History, toggle to a filter that had no existing archived threats and ensure that the filter toggles remain accessible and function accordingly

## Screenshots

History error state:
![Screen Shot on 2024-07-31 at 13-52-55](https://github.com/user-attachments/assets/c2e4b826-a110-4813-afa0-4fd23996b296)

Scanner error  state (access to Scanner/History toggles):
![Screen Shot on 2024-07-31 at 13-53-17](https://github.com/user-attachments/assets/827c0b07-ed70-4d62-b6cb-563631edf7a8)

Scanner scanning state (access to Scanner/History toggles):
![Screen Shot on 2024-07-31 at 13-53-55](https://github.com/user-attachments/assets/644be353-b720-4e54-88ff-f0722d45e7a0)

History empty filter (access to filter toggles):
![Screen Shot on 2024-07-31 at 13-52-14](https://github.com/user-attachments/assets/c1c91c29-a38a-485f-a593-aa3755324053)
